controllers/fcToolsController.js

```diff
@@ -17,73 +17,151 @@ exports.fitTool = function(req, res) {
+            } else {
+                req.flash("content", {"class":"error", "title":"Woops!", "message":"We couldn't find " + req.params.pilotname + ". Have they used our system before?"});
+                res.redirect(`/commander`);                        
+            }
+        })
+    }).catch(function (err) {
+        req.flash("content", {"class":"error", "title":"Woops!", "message":"We couldn't find " + req.params.pilotname + ". Did you spell your pilot name correctly?"});
+        res.redirect(`/commander`);
+    })
+}
+
+/*
+* AJAX Search for the pilot
+* @params req{}
+* @return res{} url for pilot page || error
+*/
+exports.searchForPilot = function(req, res){
+    if (req.isAuthenticated() && req.user.role.numeric > 0) {
+        esi.characters.search.strict(req.body.search).then(function (results) {
+            //ESI cannot find pilot
+            if(results == undefined){
+                res.status(404).send("ESI Search Fails - No pilot found");
+                return;
+            }
+    
+            users.findAndReturnUser(results[0], function(user){
+                //ESI finds pilot but they aren't registered
+                if(!user){
+                    res.status(400).send("Pilot is not registered");
+                    return;
+                }
+                res.status(200).send({"name": user.name, "url": "/commander/" + user.name.replace('/s/g', '-').replace(/\s/g, '-') + "/info"});
             })
-        } else {
-            res.redirect('/commander/'+req.user.name.replace(/\s+/g, '-')+'/skills')
-        }
+            
+        })
     } else {
         req.flash("content", {"class":"error", "title":"Not Authorised!", "message":"Only our FC team has access to that page! Think this is an error? Contact a member of leadership."});
         res.status(403).redirect("/");
     }
 }
 
-exports.waitlistLog = function(req, res) {
-    if (req.isAuthenticated() && req.user.roleNumeric > 0) {
-        wlog.getWeek(function(logData){
-            
-            for(var i = 0; i < logData.length; i++){
-                let timestamp = logData[i].time;                
-                
-                month = timestamp.getMonth()+1;
-                dt = timestamp.getDate();
-                hh = timestamp.getHours();
-                mm = timestamp.getMinutes()
 
-                if (dt < 10) {
-                dt = '0' + dt;
-                }
-                if (month < 10) {
-                month = '0' + month;
-                }
-                if (hh < 10) {
-                    hh = '0' + hh;
-                }
-                if (mm < 10) {
-                    mm = '0' + mm;
-                }
-                
-                logData[i].time = hh + ":" + mm + " " +month + '/' +dt;
+/*
+* Returns a view of the pilots skills
+* @params req{}
+* @return Renders view
+*/
+exports.skillsChecker = function(req, res) {
+    if(!req.isAuthenticated() || req.isAuthenticated() && req.user.role.numeric < 1){
```

Comment:
This seems to be a constant check on most controller methods. Can we pull this into a helper method?